src/main/java/codechicken/microblock/init/CBMicroblockModContent.java

```diff
@@ -105,311 +106,331 @@ private static void onProcessIMC(InterModProcessEvent event) {
+            registerMaterial(r, new BlockMicroMaterial(Blocks.DEEPSLATE_COPPER_ORE));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.OXIDIZED_CUT_COPPER));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.WEATHERED_CUT_COPPER));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.EXPOSED_CUT_COPPER));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.CUT_COPPER)); //TODO Oxidization (normal -> exposed -> weathered -> oxidized)
+            registerMaterial(r, new BlockMicroMaterial(Blocks.WAXED_COPPER_BLOCK));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.WAXED_WEATHERED_COPPER));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.WAXED_EXPOSED_COPPER));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.WAXED_OXIDIZED_COPPER));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.WAXED_OXIDIZED_CUT_COPPER));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.WAXED_WEATHERED_CUT_COPPER));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.WAXED_EXPOSED_CUT_COPPER));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.WAXED_CUT_COPPER));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.DRIPSTONE_BLOCK));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.ROOTED_DIRT));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.MUD));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.DEEPSLATE));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.COBBLED_DEEPSLATE));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.POLISHED_DEEPSLATE));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.DEEPSLATE_TILES));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.DEEPSLATE_BRICKS));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.CHISELED_DEEPSLATE));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.CRACKED_DEEPSLATE_BRICKS));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.CRACKED_DEEPSLATE_TILES));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.SMOOTH_BASALT));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.RAW_IRON_BLOCK));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.RAW_COPPER_BLOCK));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.RAW_GOLD_BLOCK));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.OCHRE_FROGLIGHT));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.VERDANT_FROGLIGHT));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.PEARLESCENT_FROGLIGHT));
+            registerMaterial(r, new BlockMicroMaterial(Blocks.REINFORCED_DEEPSLATE));
+        });
+    }
+
+    private static void registerMaterial(RegisterEvent.RegisterHelper<MicroMaterial> r, BlockMicroMaterial material) {
+        r.register(BlockMicroMaterial.makeMaterialKey(material.state), material);
```

Comment:
```suggestion
        r.register(material.getRegistryName(), material);
```

Reply:
I don't think `getRegistryName()` is usable until after registration.

```
public ResourceLocation getRegistryName() {
        return Objects.requireNonNull(MicroMaterialRegistry.MICRO_MATERIALS.getKey(this));
    }
```

`MICRO_MATERIALS` would not have a key until `r.register(BlockMicroMaterial.makeMaterialKey(material.state), material)` completes

Reply:
Ahh, damn. That's fine then.